File: authentication/account/status.go
package account

type Status int

// we use iota here to assign incremental integer values to the constants
// status values will have 0, 1, 2 etc.
const (
	Pending Status = iota
	Active
	Suspended
)

func (s Status) String() string {
	return [...]string{"PENDING", "ACTIVE", "SUSPENDED"}[s]
}

File: client/src/lib/proto/account/Status.ts
// Original file: ../proto/account.proto

export const Status = {
  PENDING: 'PENDING',
  ACTIVE: 'ACTIVE',
  SUSPENDED: 'SUSPENDED',
} as const;

export type Status =
  | 'PENDING'
  | 0
  | 'ACTIVE'
  | 1
  | 'SUSPENDED'
  | 2

export type Status__Output = typeof Status[keyof typeof Status]
